src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/IntegrationTestBase.kt

```diff
@@ -209,20 +213,30 @@ abstract class IntegrationTestBase {
 
   @BeforeEach
   fun clearDatabase() {
-    actionPlanRepository.deleteAll() // Will also remove all Goals and Steps due to cascade
-    timelineRepository.deleteAll() // Will also remove all TimelineEvents due to cascade
-    inductionRepository.deleteAll()
-    previousQualificationsRepository.deleteAll()
-    reviewRepository.deleteAll()
-    reviewScheduleRepository.deleteAll()
-    reviewScheduleHistoryRepository.deleteAll()
-    noteRepository.deleteAll()
-    inductionScheduleRepository.deleteAll()
-    inductionScheduleHistoryRepository.deleteAll()
-
-    clearQueues()
-
-    wiremockService.resetAllStubsAndMappings()
+    repeat(2) { attempt ->
+      // this sometimes fails - try to do this twice with a delay
+      try {
+        actionPlanRepository.deleteAll()
+        timelineRepository.deleteAll()
+        inductionRepository.deleteAll()
+        previousQualificationsRepository.deleteAll()
+        reviewRepository.deleteAll()
+        reviewScheduleRepository.deleteAll()
+        reviewScheduleHistoryRepository.deleteAll()
+        noteRepository.deleteAll()
+        inductionScheduleRepository.deleteAll()
+        inductionScheduleHistoryRepository.deleteAll()
+
+        clearQueues()
+        wiremockService.resetAllStubsAndMappings()
+
+        return
```

Comment:
I'm assuming this `return` returns out of the whole `repeat` rather than the `try` block; and assuming that it only runs twice when it fails?

Reply:
yes the return exits the 'loop' `repeat` is literally a wrapper for a for loop